pkg/controller/operators/operator_controller_test.go

```diff
@@ -77,16 +75,62 @@ var _ = Describe("Operator Controller", func() {
 			})
 
 			It("should re-create it", func() {
-				Eventually(func() error {
-					return k8sClient.Get(ctx, name, operator)
-				}).Should(Succeed())
+				// There's a race condition between this test and the controller. By the time,
+				// this function is running, we may be in one of three states.
+				//   1. The original deletion in the test setup has not yet finished, so the original
+				//      operator resource still exists.
+				//   2. The operator doesn't exist, and the controller has not yet re-created it.
+				//   3. The operator has already been deleted and re-created.
+				//
+				// This race condition means we need to test for either of two possibilities when getting
```

Comment:
Could we add a value to original resource that would change if it was recreated? This would make the check much simpler: just wait until a resource with the same name exists that has a changed "sentinel value". 

Initially, I was thinking that we could use the `meta.id` field, but since this is a unit test, I'm not sure if the behavior of the mock apiserver will be the same (i.e. id fields aren't generated or unique). If using the resource ID doesn't work, we could always add an annotation or label, then check if it no longer exists.

Reply:
This is a good idea. This is running in envtest with a real etcd and kube-apiserver, so I think the meta.id should be different. If this works, then you're right, all we need to do is wait for a different UID.